src/commons/navigationBar/subcomponents/AcademyNavigationBar.tsx

```diff
@@ -152,4 +51,86 @@ export const icons: IconName[] = [
   IconNames.CALCULATOR
 ];
 
+export const assessmentTypesToNavlinkInfo = ({
+  assessmentTypes,
+  courseId,
+  isEnrolledInACourse
+}: {
+  assessmentTypes?: string[];
+  courseId?: number;
+  isEnrolledInACourse: boolean;
+}): NavbarEntryInfo[] =>
+  (assessmentTypes || []).map((assessmentType, idx) => ({
+    to: `/courses/${courseId}/${assessmentTypeLink(assessmentType)}`,
+    icon: icons[idx],
+    text: assessmentType,
+    disabled: !isEnrolledInACourse,
+    hasNotifications: true,
+    hiddenInBreakpoints: ['xs', 'sm']
+  }));
+
+export const getStaffNavlinkInfo = ({
+  courseId,
+  role
+}: {
+  courseId?: number;
+  role?: Role;
+}): NavbarEntryInfo[] => {
+  const isStaffOrAdmin = role === Role.Admin || role === Role.Staff;
+  const isAdmin = role === Role.Admin;
+
+  return [
+    {
+      to: `/courses/${courseId}/groundcontrol`,
+      icon: IconNames.SATELLITE,
+      text: 'Ground Control',
+      disabled: !isStaffOrAdmin,
+      hiddenInBreakpoints: ['xs', 'sm']
+    },
+    {
+      to: `/courses/${courseId}/dashboard`,
+      icon: IconNames.GLOBE,
+      text: 'Dashboard',
+      disabled: !isStaffOrAdmin,
+      hiddenInBreakpoints: ['xs', 'sm']
+    },
+    {
+      to: `/courses/${courseId}/sourcereel`,
+      icon: IconNames.MOBILE_VIDEO,
+      text: 'Sourcereel',
+      disabled: !isStaffOrAdmin,
+      hiddenInBreakpoints: ['xs', 'sm', 'md']
+    },
+    {
+      to: `/courses/${courseId}/grading`,
+      icon: IconNames.ENDORSED,
+      text: 'Grading',
+      disabled: !isStaffOrAdmin,
+      hasNotifications: true,
+      hiddenInBreakpoints: ['xs', 'sm', 'md']
+    },
+    {
+      to: `/courses/${courseId}/storysimulator`,
+      icon: IconNames.CROWN,
+      text: 'Story Simulator',
+      disabled: !isStaffOrAdmin,
+      hiddenInBreakpoints: ['xs', 'sm', 'md']
+    },
+    {
+      to: `/courses/${courseId}/xpcalculation`,
+      icon: IconNames.CALCULATOR,
+      text: 'XP Calculation',
+      disabled: !isAdmin,
+      hiddenInBreakpoints: ['xs', 'sm', 'md', 'lg']
```

Comment:
Likewise, originally, XP Calculation was only hidden in the `xs` and `sm` breakpoints. Is this change intentional as part of the improvements to readability of NavBar? I'm not sure if this refers to code readability of UX improvements. It definitely does improve the code readability, but in terms of the UX, was this an intentional change? Just making sure haha